src/interpret/interpret.ml

```diff
@@ -556,8 +549,8 @@ struct
       Stack.t -> (f, r) Extern_func.atype -> f -> r Choice.t =
      fun stack ty f ->
       match ty with
-      | Mem args ->
-        let* mem = Env.get_memory env mem_0 in
+      | Mem (memid, args) ->
+        let* mem = Env.get_memory env (Int32.to_int memid) in
```

Comment:
I think we can safely change this to be an `int` to avoid the conversion. And it'll be closer to what is done at the instruction level where in `Memory_size memid` for instance, `memid` is an `int`.

Reply:
I agree, done!